src/OrchardCore/OrchardCore.Abstractions/Shell/Extensions/ShellHostExtensions.cs

```diff
@@ -50,5 +53,13 @@ public static ShellSettings GetSettings(this IShellHost shellHost, string tenant
 
             return settings;
         }
+
+        /// <summary>
+        /// Retrieves all shell settings with the given state
+        /// </summary>
+        public static IEnumerable<ShellSettings> GetAllSettings(this IShellHost shellHost, TenantState tenantState)
+        {
+            return shellHost.GetAllSettings().Where(x => x.State == tenantState);
```

Comment:
Maybe `GetSettingsByState()`

The comments could be `/// Retrieves the shell settings being in the specified state.` <= Use a dot at the end.

In the delegate at least use `settings` in place of `x`.